functionalDB/chapter.md

```diff
@@ -726,42 +731,40 @@ A clause in a query is composed of three predicates. The following table defines
 
-Engineering is all about managing tradeoffs, and designing our query engine is no different. In our case, the first tradeoff we must make is feature-richness versus complexity. Resolving this tradeoff requires us to look at common use-cases of the system, and from there deciding on what limitations would be acceptable. 
-
-In our database, the decision was to build a query engine with the following limitations:
+In our database, we decided to build a query engine with the following limitations:
 
 * Users cannot define logical operations between the clauses; they are always ‘ANDed’ together. (This can be worked around by using unary or binary predicates.)
 * If there is more than one clause in a query, there must be one variable that is found in all of the clauses of that query. This variable acts as a joining variable. This limitation simplifies the query optimizer.
 * A query is only executed on a single database. 
 
-While these design decisions result in a query language that is less rich than Datalog, we are still able to support many types of simple-but-useful queries.
+While these design decisions result in a query language that is less rich than Datalog, we are still able to support many types of simple but useful queries.
 
-### Query engine design
+### Query Engine Design
 
-While our query language allows the user to specify _what_ they want to access, it hides the details of _how_ this will be accomplished. The *query engine* is the database component responsible for yielding the data for a given query. 
+While our query language allows the user to specify _what_ they want to access, it hides the details of _how_ this will be accomplished. The `query engine` is the database component responsible for yielding the data for a given query. 
 
 This involves four steps:
 
-1. Transformation to internal representation: transform the query from its textual form into a data structure that is consumed by the query planner.
+1. Transformation to internal representation: Transform the query from its textual form into a data structure that is consumed by the query planner.
 2. Building a query plan: Determine an efficient _plan_ for yielding the results of the given query. In our case, a query plan is a function to be invoked.
 3. Executing the plan: Execute the plan and send its results to the next phase.
 4. Unification and reporting: Extract only the results that need to be reported and format them as specified.
 
-#### Phase 1 - Transformation
+#### Phase 1: Transformation
 
 In this phase, we transform the given query from a representation that is easy for the user to understand into a representation that can be consumed efficiently by the query planner. 
 
-The *:find* part of the query is transformed into a set of the given variable names:
+The `:find` part of the query is transformed into a set of the given variable names:

```

Comment:
How about adding a sentence before this line that says something like "A query has two parts, the :find part and the :where part, each one transforms in a different way"